packages/server/src/utils/workflow.ts

```diff
@@ -214,3 +138,62 @@ async function downloadLlamaCloudFilesFromNodes(nodes: SourceEventNode[]) {
     downloadedFiles.push(node.filePath);
   }
 }
+
+export async function* toStreamGenerator(
+  input: AsyncIterable<ChatResponseChunk> | string,
+): AsyncGenerator<ChatResponseChunk> {
+  if (typeof input === "string") {
+    yield { delta: input } as ChatResponseChunk;
+    return;
+  }
+
+  for await (const chunk of input) {
+    yield chunk;
+  }
+}
+
+function workflowToEngineResponseStream(
+  workflowStream: AsyncIterable<WorkflowEventData<unknown>>,
+  dataStream: StreamData,
+): ReadableStream<EngineResponse> {
+  const { readable, writable } = new TransformStream<
```

Comment:
a `TransformStream` is created by overwriting the `transform(chunk, controller)` method. here we want to transform the stream of `WorkflowEventData<unknown>` to `StreamData` (by not using `LlamaIndexAdapter` as it only works with `EngineResponse`)